src/ui/pages/SendForm/SuccessState/SuccessState.tsx

```diff
@@ -13,29 +12,42 @@ import { useActionStatusByHash } from 'src/ui/shared/forms/SuccessState/useActio
 import { SuccessStateNft } from 'src/ui/shared/forms/SuccessState/SuccessStateNft';
 import { NavigationTitle } from 'src/ui/components/NavigationTitle';
 import { GasbackDecorated } from 'src/ui/components/GasbackDecorated';
+import type { AddressPosition } from 'defi-sdk';
+import type { SendFormState } from '../SendForm';
+import { useCurrentPosition } from '../shared/useCurrentPosition';
+import type { NftId } from '../shared/useNftPosition';
+import { useNftPosition } from '../shared/useNftPosition';
 
 export interface SendFormSnapshot {
   state: SendFormState;
-  tokenItem: SendFormView['tokenItem'];
-  nftItem: SendFormView['nftItem'];
 }
 
 export function SuccessState({
+  address,
   sendFormSnapshot,
+  positions,
   gasbackValue,
   hash,
   onDone,
 }: {
+  address: string;
   sendFormSnapshot: SendFormSnapshot;
+  positions: AddressPosition[];
   gasbackValue: number | null;
   hash: string;
   onDone: () => void;
 }) {
   const { networks } = useNetworks();
-  const { tokenItem, nftItem, state } = sendFormSnapshot;
-  const { type, tokenChain, nftChain, to } = state;
-  const currentChain = type === 'token' ? tokenChain : nftChain;
-  invariant(to && currentChain, 'Required Form values are missing');
+  const { state: formState } = sendFormSnapshot;
+  const tokenItem = useCurrentPosition(formState, positions);
+  const { type, tokenChain, to, nftId } = formState;
+  const chain = tokenChain ? createChain(tokenChain) : null;
+  invariant(to && chain, 'Required Form values are missing');
+  const { value: nftItem } = useNftPosition({
```

Comment:
is it reliable to fetch nft position from backend after it was sent?
Probably is will be still here because of the delay, but not sure about rejecting snapshot here.

Reply:
Good point, let me think about this
It's supposed to be in local cache anyway